burnman/layer.py

```diff
@@ -0,0 +1,797 @@
+        pressure_top : float
+        Pressure (Pa) at the top of the layer. 
+        gravity_bottom : float
+        gravity (m/s^2) at the bottom the layer
+        n_max_iterations : int
+        Maximum number of iterations to reach self-consistent pressures (default = 50)
+        """
+        self.reset()
+        assert(pressure_mode == 'user_defined' or pressure_mode == 'selfconsistent')
+        assert(self.temperature_mode is not None)
+
+        self.gravity_bottom = gravity_bottom
+
+        if pressure_mode == 'user_defined':
+            assert(len(pressures) == len(self.depths))
+            self._pressures = pressures
+            warnings.warn(
+                "By setting the pressures in Layer it is unlikely to be self-consistent")
+            self._temperatures = self._evaluate_temperature(
+                self._pressures, self.temperature_top)
+
+        if pressure_mode == 'selfconsistent':
+            self.pressure_top = pressure_top
+            ref_press = np.zeros_like(pressures)
+            new_press = self.pressure_top + \
+                (self.depths - min(self.depths)) * \
+                2.e5  # initial pressure curve guess
+            temperatures = self._evaluate_temperature(
+                new_press, self.temperature_top)
+            # Make it self-consistent!!!
+            i = 0
+
+            while i < n_max_iterations:
+                i += 1
+                ref_press = new_press
+                new_grav, new_press = self._evaluate_eos(
+                    new_press, temperatures, gravity_bottom, pressure_top)
+                temperatures = self._evaluate_temperature(
+                    new_press, self.temperature_top)
+                rel_err = abs(
+                    (max(ref_press) - max(new_press)) / max(new_press))
+                if self.verbose:
+                    print(
+                        "Iteration %i  maximum core pressure error between iterations: %e" %
+                        (i, rel_err))
+
+                if rel_err < 1e-5:
+                    break
+
+            self._pressures = new_press
+            self._temperatures = temperatures
+
+        self.sublayers = []
+        for l in range(len(self.depths)):
+            self.sublayers.append(self.material.copy())
+            self.sublayers[l].set_state(
+                self._pressures[l], self._temperatures[l])
+
+    def _evaluate_temperature(self, pressures=None, temperature_top=None):
+        """
+        Returns the temperatures of the layer for given pressures.
+        Used by set_state()
+        """
+        if self.temperature_mode == 'adiabat' or self.temperature_mode == 'modified_adiabat':
+            adiabat = geotherm.adiabatic(
+                pressures, temperature_top, self.material)
+        else:
+            adiabat = np.zeros_like(self.depths)
+        return adiabat + self.usertemperatures
+
+    def _evaluate_eos(self, pressures, temperatures, gravity_bottom, pressure_top):
```

Comment:
I think the choice of gravity_bottom and pressure_top needs some documentation, both here and briefly in the Class blurb.